### PR TITLE
Add `streamName` and epxlicitly bind to consumer and stream

### LINKED ISSUE #2243: [NATS JetStream] Durable Consumer deleted after closing the app
## Expected Behavior

Durable consumers are not deleted.

## Actual Behavior

Durable consumers are deleted when app closes.

## Steps to Reproduce the Problem

Call an app that Subscribes to JetStream.
Gracefully shutdown the app.

## Release Note

RELEASE NOTE:

### PR DESCRIPTION
**Pull request by Jarema**
# Description

_Please explain the changes you've made_

## Issue reference

This addesses #2243 and moves to more explicit API that allows to control lifecycle of NATS JetStream Consumers in DAPR.

closes #2243

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Review by ItalyPaleAle — DISMISSED** [review]
Thanks for this contribution!

Can you please document the new metadata option `streamName` in https://github.com/dapr/docs ?

**Comment by Jarema** [comment]
@ItalyPaleAle sure, will issuse a PR to the docs.

**Review by NickLarsenNZ — DISMISSED** [review]
Looks good

**Comment by yaron2** [comment]
@Jarema please fix the conflicts and we can merge. Thanks!

File: pubsub/jetstream/jetstream.go [hunk 1]
```diff
@@ -176,14 +179,27 @@ func (js *jetstreamPubSub) Subscribe(ctx context.Context, req pubsub.SubscribeRe
 	}
 
 	var err error
+	streamName := js.meta.streamName
+	if streamName == "" {
+		streamName, err = js.jsc.StreamNameBySubject(req.Topic)
```
[the remark this answers is not in the document]
Comment (yaron2) [reply pubsub/jetstream/jetstream.go:184]: Once we updated Dapr's master branch with this version, you should be able to get a nightly build that incorporates this change.

[the remark this answers is not in the document]
Comment (NickLarsenNZ) [reply pubsub/jetstream/jetstream.go:184]: Ok cool, I'll assume it will come into dapr 1.10

**Review by berndverst — DISMISSED** [review]

**Comment by berndverst** [comment]
I cleaned up the conflicts and bumped the version to the latest tag `1.19.1`

**Review by berndverst — APPROVED** [review]

**Pull request merged**